include/linux/tee_drv.h

```diff
@@ -98,14 +102,19 @@ struct tee_driver_ops {
 	void (*get_version)(struct tee_device *teedev,
 			    struct tee_ioctl_version_data *vers);
 	int (*open)(struct tee_context *ctx);
+	void (*pre_release)(struct tee_context *ctx);
 	void (*release)(struct tee_context *ctx);
 	int (*open_session)(struct tee_context *ctx,
 			    struct tee_ioctl_open_session_arg *arg,
-			    struct tee_param *param);
+			    struct tee_param *normal_param,
+			    u32 num_normal_params,
+			    struct tee_param *ocall_param);
```

Comment:
This will break bisectability unless the users are minimally updated too (in the same commit). I think this should be avoided as much as possible.

Reply:
I agree. The change here in this commit "tee: OCALL support" should include minimal changes in the optee and amdtee drivers to make things compile and work (obviously without OCALL support in the drivers yet).

Reply:
Sorry, that was silly on my part. Fixed.